### PR TITLE
Update Pattern Lab Schema Validation Syntax = ~57% faster compiles!

### PR DESCRIPTION
**Pull request by sghoweri**
Based on some findings from @joekarasek on our Pattern Lab build performance times, it looks like the syntax we had been using is causing quite a performance hit (due to constantly reading / writing from the file system with every single PL compile).

As a result, this PR goes in and globally updates the Twig template schema validation syntax so we reference the schema data already loaded into memory, resulting in ~57% faster compile times (or nearly 5s shaved off with every single compile)! 💯 

Side note: this PR also makes a few fixes necessary to get these updates to compile as expected and not throw any errors (including adding missing package.json file keys + workarounds to a few problematic schemas still getting worked on)

### CODE REVIEW

**Comment by sghoweri** [comment]
@remydenton It also seems like Travis now compiles PL [in about 30 seconds](https://travis-ci.org/bolt-design-system/bolt/builds/379244460#L1333) instead of [around 114 seconds](https://travis-ci.org/bolt-design-system/bolt/builds/378783614#L1335) as it had been previously

File: packages/components/bolt-band/src/band.twig [hunk 1]
```diff
@@ -65,5 +65,5 @@
 </bolt-{{ componentName }}>
 
 {% if enable_json_schema_validation %}
-  {{ validate_data_schema('@bolt-components-band/band.schema.yml', _self) | raw }}
+  {# {{ validate_data_schema(bolt.data.components['@bolt-components-band'].schema, _self) | raw }} #}
```
Comment (remydenton) [inline packages/components/bolt-band/src/band.twig]: Is there a reason this is commented out now?

File: packages/components/bolt-navbar/navbar.twig [hunk 2]
```diff
@@ -56,5 +56,5 @@
 
 
 {% if enable_json_schema_validation %}
-  {{ validate_data_schema('@bolt-components-navbar/navbar.schema.yml', _self) }}
-{% endif %}
+  {{ validate_data_schema(bolt.data.components['@bolt-components-navbar'].schema, _self) | raw }}
+{% endif %}
```
Comment (remydenton) [inline packages/components/bolt-navbar/navbar.twig]: Newline at the end of the file is now missing.

File: packages/components/bolt-headline/src/_typography.twig [hunk 3]
```diff
@@ -76,5 +76,5 @@
 </{{ tag }}>
 
 {% if enable_json_schema_validation %}
-  {{ validate_data_schema('@bolt-components-headline/headline.schema.yml', _self) | raw }}
-{% endif %}
+  {{ validate_data_schema(bolt.data.components['@bolt-components-headline'].schema, _self) | raw }}
+{% endif %}
```
Comment (remydenton) [inline packages/components/bolt-headline/src/_typography.twig]: Newline at the end of the file is now missing.

File: packages/components/bolt-dropdown/dropdown.schema.yml [hunk 4]
```diff
@@ -13,6 +13,6 @@ properties:
     type: boolean
     description: "`True` to hide toggle until mobile screen size"
     default: false
-  content:
-    type: any
-    description: "**All of the items in the dropdown** -- generally works by including `@bolt-components-nav/nav.twig` with `links` array of objects containing `text` & `url`"
\ No newline at end of file
+  # content:
+  #   type: any
+  #   description: "**All of the items in the dropdown** -- generally works by including `@bolt-components-nav/nav.twig` with `links` array of objects containing `text` & `url`"
```
Comment (remydenton) [inline packages/components/bolt-dropdown/dropdown.schema.yml]: I thought this commit was just changing syntax ;)

How come the schema was actually changed here?

File: packages/components/bolt-tooltip/tooltip.schema.yml [hunk 5]
```diff
@@ -47,9 +47,9 @@ properties:
           name:
             type: string
             description: The icon for our toggle
-  content:
-    type: any
-    description: This can be text OR an included bolt component (like Block List)
+  # content:
+  #   type: any
+  #   description: This can be text OR an included bolt component (like Block List)
```
Comment (remydenton) [inline packages/components/bolt-tooltip/tooltip.schema.yml]: As with dropdown, how come this is commented out now?

File: packages/components/bolt-band/src/band.twig (hunk 1 above)
Comment (sghoweri) [reply packages/components/bolt-band/src/band.twig]: Let me verify but I feel like this schema was throwing an error (due to it not getting used till I made these changes?) - lemme double check!

File: packages/components/bolt-dropdown/dropdown.schema.yml (hunk 4 above)
Comment (sghoweri) [reply packages/components/bolt-dropdown/dropdown.schema.yml]: I’ll also double check this one but I THINK the “any” prop type isn’t working for some reason so this schema breaks when it’s actually wired up to be used

File: packages/components/bolt-navbar/navbar.twig (hunk 2 above)
Comment (sghoweri) [reply packages/components/bolt-navbar/navbar.twig]: Fixed! (went through all the components and normalized the placement of the `validate_data_schema` call + added an EOF line if it was missing)

File: packages/components/bolt-dropdown/dropdown.schema.yml (hunk 4 above)
Comment (sghoweri) [reply packages/components/bolt-dropdown/dropdown.schema.yml]: @remydenton I had to add the "any" schema prop type to our global list of allowed values -- previously (before making this latest update) uncommenting those 3 lines would throw an error and not allow the tools to boot up.

File: packages/components/bolt-tooltip/tooltip.schema.yml (hunk 5 above)
Comment (sghoweri) [reply packages/components/bolt-tooltip/tooltip.schema.yml]: Same issue with the previously mentioned "any" prop type -- uncommenting this should now work as expected.

**Comment by sghoweri** [comment]
> I just discovered some documentation that needs to be updated as well:
https://bolt-design-system.com/docs/api/twig/bolt-core.html#validate_data_schema-path-_self-

@remydenton good catch! Updating this as part the updates I'm about to push up

**Comment by sghoweri** [comment]
@remydenton can I get you to sign off on this? This should be good to go with the latest updates just pushed up!

**Comment by sghoweri** [comment]
@remydenton @theSadowski @mikemai2awesome @rockymountainhigh1943 @joekarasek 

I’m going to cut the cord on getting this merged in and per my earlier comment (https://github.com/bolt-design-system/bolt/pull/696/#issuecomment-393685057) recommend we globally update the schema validation to be at the top of each component’s Twig file after this is merged. #babySteps

**Comment by remydenton** [comment]
One way to avoid this: **don't slip in new changes after your PR has been reviewed the first time around**.  If you just update the things that are requested to be fixed, it should be very easy for the reviewer to quickly review a second time and hit the approve button.  If you then want to do something new like`normalize the validate_data_schema check to occur in the same location in every component`, _you_ should be creating a separate PR for that, not me.

The real source of the problem here is that there was slow turnaround on both PR review and code changes following the review--  i.e. you and I are both to blame for letting too much time elapse.  Let's resolve to make that better!

**Comment by remydenton** [comment]
Well, looking back, I reviewed it the day you created the PR, and it took 15 days for you to address the requested changes.  So I don't think this is a genuine excuse for approving your own code:

> I’m not ok having the team continue to wait on the speed boost we should see from this 

Anyway, the principle stands-- keep the turnaround on both code review and code changes snappy and we won't have this issue.

**Comment by remydenton** [comment]
Sorry, one more post :) . I just wanted to acknowledge that I took three days to review your feedback, so this indeed on my shoulders too (and my last comment is unfair, apologies).  Long story short, turnaround time is critical and I resolve to prioritize it!

**Pull request merged**